SMF 2.1/SearchAPI-Sphinxql.php

```diff
@@ -168,6 +162,28 @@ public static function searchSettings(&$config_vars)
 		else
 			$context['sphinx_version'] = '3.0';
 
+		// Change settings for older sphinxs
+		if (!empty($context['sphinx_version']) && version_compare($context['sphinx_version'], '3.5', '>'))
+		{
+			$part1 = array_slice($local_config_vars, 0, 3);
+			$part2 = array_slice($local_config_vars, 4, 5);
+			$part3 = array_slice($local_config_vars, 7);
+			
+			$local_config_vars = array_merge(
+				$part1,
+				array(
+					array('text', 'sphinx_data_path', 65, 'default_value' => '/var/sphinx/data', 'subtext' => $txt['sphinx_data_path_subtext']),
+					array('text', 'sphinx_log_path', 65, 'default_value' => '/var/sphinx/log', 'subtext' => $txt['sphinx_log_path_subtext']),
+					array('text', 'sphinx_stopword_path', 65, 'default_value' => '', 'subtext' => $txt['sphinx_stopword_path_subtext']),
+				),
+				$part2,
+				$part3
+			);
+		}
```

Comment:
perhaps you could add named keys to the old settings and `unset()` them as needed. Easier to maintain in the future and no index juggling should they ever be rearranged.